File: tests/spring/library/src/main/java/com/example/library/web/LibraryExceptionHandler.java
package com.example.library.web;

import com.example.library.service.BookNotFoundException;
import com.example.library.service.InvalidBookException;
import org.springframework.http.HttpStatus;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;

@ControllerAdvice
public class LibraryExceptionHandler {

    @ExceptionHandler(BookNotFoundException.class)
    @ResponseStatus(HttpStatus.NOT_FOUND)
    public String notFound(BookNotFoundException e, Model model) {
        model.addAttribute("message", e.getMessage());
        return "error/404";
    }

    @ExceptionHandler(InvalidBookException.class)
    @ResponseStatus(HttpStatus.BAD_REQUEST)
    public String invalid(Model model) {
        model.addAttribute("message", "invalid book");
        return "error/404";
    }
}

File: tests/spring/library/expected/src/web/library_exception_handler.rs
//! Translated from `LibraryExceptionHandler` (LibraryExceptionHandler.java) by Java2RustTranslator.

use axum::http::StatusCode;
use axum::response::{IntoResponse, Response};

use crate::error::Error;
use crate::views::Error404View;

/// `@ExceptionHandler(BookNotFoundException)`
pub fn not_found(e: &Error) -> Response {
    let model_message = e.to_string();
    (StatusCode::NOT_FOUND, Error404View {
        message: Some(model_message),
    }).into_response()
}

/// `@ExceptionHandler(InvalidBookException)`
pub fn invalid() -> Response {
    let model_message = "invalid book".to_string();
    (StatusCode::BAD_REQUEST, Error404View {
        message: Some(model_message),
    }).into_response()
}
